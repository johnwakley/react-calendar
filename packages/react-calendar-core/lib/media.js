'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.media = exports.mediaWithBreakpoints = undefined;

var _templateObject = _taggedTemplateLiteral(['\n      @media (max-width: ', 'em) {\n        ', '\n      }\n    '], ['\n      @media (max-width: ', 'em) {\n        ', '\n      }\n    ']);

var _styledComponents = require('styled-components');

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); } /**
                                                                                                                                                   * MIT License
                                                                                                                                                   *
                                                                                                                                                   * Copyright (c) 2017 johnwakley
                                                                                                                                                   *
                                                                                                                                                   * Permission is hereby granted, free of charge, to any person obtaining a copy
                                                                                                                                                   * of this software and associated documentation files (the "Software"), to deal
                                                                                                                                                   * in the Software without restriction, including without limitation the rights
                                                                                                                                                   * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
                                                                                                                                                   * copies of the Software, and to permit persons to whom the Software is
                                                                                                                                                   * furnished to do so, subject to the following conditions:
                                                                                                                                                   *
                                                                                                                                                   * The above copyright notice and this permission notice shall be included in all
                                                                                                                                                   * copies or substantial portions of the Software.
                                                                                                                                                   *
                                                                                                                                                   * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
                                                                                                                                                   * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
                                                                                                                                                   * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
                                                                                                                                                   * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
                                                                                                                                                   * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
                                                                                                                                                   * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
                                                                                                                                                   * SOFTWARE.
                                                                                                                                                   *
                                                                                                                                                   */

var defaultSizes = {
  mobile: 640,
  tablet: 1280,
  desktop: 1920
};

var mediaWithBreakpoints = exports.mediaWithBreakpoints = function mediaWithBreakpoints() {
  var breakpoints = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultSizes;

  var breaks = Object.assign({}, defaultSizes, breakpoints);

  // Iterate through the breakpoints and create a media template
  return Object.keys(breaks).reduce(function (acc, label) {
    acc[label] = function () {
      return (0, _styledComponents.css)(_templateObject, breaks[label] / 16, _styledComponents.css.apply(undefined, arguments));
    };

    return acc;
  }, {});
};

var media = exports.media = mediaWithBreakpoints();