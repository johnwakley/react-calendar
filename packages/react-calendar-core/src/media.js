/**
 * MIT License
 *
 * Copyright (c) 2017 johnwakley
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 *
 */

// @flow

import { css } from 'styled-components'

export type MediaBreakpoints = {
  mobile: number,
  tablet: number,
  desktop: number
}

const defaultSizes: MediaBreakpoints = {
  mobile: 640,
  tablet: 1280,
  desktop: 1920
}

export const mediaWithBreakpoints = (
  breakpoints: MediaBreakpoints = defaultSizes
): any => {
  const breaks = Object.assign({}, defaultSizes, breakpoints)

  // Iterate through the breakpoints and create a media template
  return Object.keys(breaks).reduce((acc, label) => {
    acc[label] = (...args) => css`
      @media (max-width: ${breaks[label] / 16}em) {
        ${css(...args)}
      }
    `

    return acc
  }, {})
}

export const media = mediaWithBreakpoints()
