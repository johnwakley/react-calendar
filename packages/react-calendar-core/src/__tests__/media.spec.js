import {media, mediaWithBreakpoints, MediaBreakpoints} from '../media'

const customSizes: MediaBreakpoints = {
  mobile: 100,
  tablet: 200,
  desktop: 300
}

const customMedia = mediaWithBreakpoints(customSizes)

describe('media', () => {
  it('should have default mobile breakpoint', () => {
    const style = media.mobile`background: papayawhip;`
    expect(style).toMatchSnapshot()
  })

  it('should have default tablet breakpoint', () => {
    const style = media.tablet`background: papayawhip;`
    expect(style).toMatchSnapshot()
  })

  it('should have default desktop breakpoint', () => {
    const style = media.desktop`background: papayawhip;`
    expect(style).toMatchSnapshot()
  })

  it('should have custom mobile breakpoint', () => {
    const style = customMedia.mobile`background: papayawhip;`
    expect(style).toMatchSnapshot()
  })

  it('should have custom tablet breakpoint', () => {
    const style = customMedia.tablet`background: papayawhip;`
    expect(style).toMatchSnapshot()
  })

  it('should have custom desktop breakpoint', () => {
    const style = customMedia.desktop`background: papayawhip;`
    expect(style).toMatchSnapshot()
  })
})
