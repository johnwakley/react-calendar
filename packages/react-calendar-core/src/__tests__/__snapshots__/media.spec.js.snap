// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media should have custom desktop breakpoint 1`] = `
Array [
  "
      @media (max-width: ",
  "18.75",
  "em) {
        ",
  "background: papayawhip;",
  "
      }
    ",
]
`;

exports[`media should have custom mobile breakpoint 1`] = `
Array [
  "
      @media (max-width: ",
  "6.25",
  "em) {
        ",
  "background: papayawhip;",
  "
      }
    ",
]
`;

exports[`media should have custom tablet breakpoint 1`] = `
Array [
  "
      @media (max-width: ",
  "12.5",
  "em) {
        ",
  "background: papayawhip;",
  "
      }
    ",
]
`;

exports[`media should have default desktop breakpoint 1`] = `
Array [
  "
      @media (max-width: ",
  "120",
  "em) {
        ",
  "background: papayawhip;",
  "
      }
    ",
]
`;

exports[`media should have default mobile breakpoint 1`] = `
Array [
  "
      @media (max-width: ",
  "40",
  "em) {
        ",
  "background: papayawhip;",
  "
      }
    ",
]
`;

exports[`media should have default tablet breakpoint 1`] = `
Array [
  "
      @media (max-width: ",
  "80",
  "em) {
        ",
  "background: papayawhip;",
  "
      }
    ",
]
`;
